vb6/VisualBasic6Lexer.g4

```diff
@@ -0,0 +1,529 @@
+
+OPTION_PRIVATE_MODULE:
+	O P T I O N ' ' P R I V A T E ' ' M O D U L E;
+
+OR: O R;
+
+OUTPUT: O U T P U T;
+
+PARAMARRAY: P A R A M A R R A Y;
+
+PRESERVE: P R E S E R V E;
+
+PRINT: P R I N T;
+
+PRIVATE: P R I V A T E;
+
+PROPERTY_GET: P R O P E R T Y ' ' G E T;
+
+PROPERTY_LET: P R O P E R T Y ' ' L E T;
+
+PROPERTY_SET: P R O P E R T Y ' ' S E T;
+
+PUBLIC: P U B L I C;
+
+PUT: P U T;
+
+RANDOM: R A N D O M;
+
+RANDOMIZE: R A N D O M I Z E;
+
+RAISEEVENT: R A I S E E V E N T;
+
+READ: R E A D;
+
+READ_WRITE: R E A D ' ' W R I T E;
+
+REDIM: R E D I M;
+
+REM: R E M;
+
+RESET: R E S E T;
+
+RESUME: R E S U M E;
+
+RETURN: R E T U R N;
+
+RMDIR: R M D I R;
+
+RSET: R S E T;
+
+SAVEPICTURE: S A V E P I C T U R E;
+
+SAVESETTING: S A V E S E T T I N G;
+
+SEEK: S E E K;
+
+SELECT: S E L E C T;
+
+SENDKEYS: S E N D K E Y S;
+
+SET: S E T;
+
+SETATTR: S E T A T T R;
+
+SHARED: S H A R E D;
+
+SINGLE: S I N G L E;
+
+SPC: S P C;
+
+STATIC: S T A T I C;
+
+STEP: S T E P;
+
+STOP: S T O P;
+
+STRING: S T R I N G;
+
+SUB: S U B;
+
+TAB: T A B;
+
+TEXT: T E X T;
+
+THEN: T H E N;
+
+TIME: T I M E;
+
+TO: T O;
+
+TRUE: T R U E;
+
+TYPE: T Y P E;
+
+TYPEOF: T Y P E O F;
+
+UNLOAD: U N L O A D;
+
+UNLOCK: U N L O C K;
+
+UNTIL: U N T I L;
+
+VARIANT: V A R I A N T;
+
+VERSION: V E R S I O N;
+
+WEND: W E N D;
+
+WHILE: W H I L E;
+
+WIDTH: W I D T H;
+
+WITH: W I T H;
+
+WITHEVENTS: W I T H E V E N T S;
+
+WRITE: W R I T E;
+
+XOR: X O R;
+
+// symbols
+
+AMPERSAND: '&';
+
+ASSIGN: ':=';
+
+AT: '@';
+
+COLON: ':';
+
+COMMA: ',';
+
+DIV: '\\' | '/';
+
+DOLLAR: '$';
+
+DOT: '.';
+
+EQ: '=';
+
+EXCLAMATIONMARK: '!';
+
+GEQ: '>=';
+
+GT: '>';
+
+HASH: '#';
+
+LEQ: '<=';
+
+LBRACE: '{';
+
+LPAREN: '(';
+
+LT: '<';
+
+MINUS: '-';
+
+MINUS_EQ: '-=';
+
+MULT: '*';
+
+NEQ: '<>';
+
+PERCENT: '%';
+
+PLUS: '+';
+
+PLUS_EQ: '+=';
+
+POW: '^';
+
+RBRACE: '}';
+
+RPAREN: ')';
+
+SEMICOLON: ';';
+
+L_SQUARE_BRACKET: '[';
+
+R_SQUARE_BRACKET: ']';
+
+// literals
+
+STRINGLITERAL: '"' (~ ["\r\n] | '""')* '"';
+
+DATELITERAL: HASH (~ [#\r\n])* HASH;
+
+COLORLITERAL: '&H' [0-9A-F]+ AMPERSAND?;
+
+INTEGERLITERAL: (PLUS | MINUS)? [0-9]+ (E INTEGERLITERAL)* (
+		HASH
+		| AMPERSAND
+		| EXCLAMATIONMARK
+		| AT
+	)?;
+
+DOUBLELITERAL: (PLUS | MINUS)? [0-9]* DOT [0-9]+ (
+		E (PLUS | MINUS)? [0-9]+
+	)* (HASH | AMPERSAND | EXCLAMATIONMARK | AT)?;
+
+FILENUMBER: HASH LETTERORDIGIT+;
+
+OCTALLITERAL: (PLUS | MINUS)? '&O' [0-7]+ AMPERSAND?;
+
+// misc
+FRX_OFFSET: COLON [0-9A-F]+;
+
+GUID:
+	LBRACE [0-9A-F]+ MINUS [0-9A-F]+ MINUS [0-9A-F]+ MINUS [0-9A-F]+ MINUS [0-9A-F]+ RBRACE;
+
+// identifier
+
+IDENTIFIER: LETTER LETTERORDIGIT*;
+
+// whitespace, line breaks, comments, ...
+
+LINE_CONTINUATION: ' ' '_' '\r'? '\n' -> skip;
+
+NEWLINE: WS? ('\r'? '\n' | COLON ' ') WS?;
+
+COMMENT:
+	WS? ('\'' | COLON? REM ' ') (
+		LINE_CONTINUATION
+		| ~ ('\n' | '\r')
+	)* -> skip;
+
+WS: [ \t]+;
+
+// letters
+
+fragment LETTER:
+	[a-zA-Z_äöüÄÖÜáéíóúÁÉÍÓÚâêîôûÂÊÎÔÛàèìòùÀÈÌÒÙãẽĩõũÃẼĨÕŨçÇ];
```

Comment:
Please fix the file encoding. I'm quite sure here should be other letters.

Reply:
The file was UTF-8 from the start and I really have no idea from which format they are. If you can help with that, I would be very thankful.

Reply:
not an encoding problem, it just listed some letter here, can be replaced with `\p{L}` (tested in VBA, it has Unicode support)

Reply:
Okay... will change VBA